### PR TITLE
Updating the roadmap to reflect the correct status

### PR DESCRIPTION
**Pull request by avishnu**
Changes in this pull request:
- Renamed the existing columns in the roadmap table, added new columns - engine type and status
- Removed the backlogs previously marked 'out-of-scope' and other backlogs which also exist out of scope as per the VISION file
- Considered the roadmap starting 2024 onwards
- Re-worded titles and description for few backlogs for better clarity
- Ordered the backlogs as per ascending time

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -6,91 +6,44 @@ Community and contributor involvement is vital for successfully implementing all
...
+
+OpenEBS follows a release cadence with a new minor release every 3-4 months.
+
+_Note_: The planned release timelines, version numbers and feature priorities are subject to change as the project maintainers/leadership/community continuously update and adjust in response to K8s industry movements, trends and our community influence.
+
+_Note 2_: The table contains the high-level backlog items. For a comprehensive list of features and issues, please refer to https://github.com/openebs/openebs/releases.
+
+| Feature | Description | Local or Replicated | Release timeline | Status |
+| :------ | :---------- | :------------------ | :--------------- | :----- |
+| One installer | Unified Helm installer for all engines, deprecates operator yaml | All | v4.0 (Q1 2024) | Completed |
```
Comment (blaisedias) [inline ROADMAP.md:20]: nit: One installer -> Single installer

File: ROADMAP.md [hunk 2]
```diff
@@ -6,91 +6,44 @@ Community and contributor involvement is vital for successfully implementing all
...
+| One documentation | Unified and restructured documentation website, deprecates mayastor.gitbook.io | All | v4.0 (Q1 2024) | Completed |
```
Comment (blaisedias) [inline ROADMAP.md:21]: nit: One documentation -> Unified documentation

File: ROADMAP.md [hunk 3]
```diff
@@ -6,91 +6,44 @@ Community and contributor involvement is vital for successfully implementing all
...
+| Volume resize | Able to increase volume size and overlaying filesystem size with I/O continuity | Replicated PV Mayastor | v4.0 (Q1 2024) | Completed |
+| Multi-replica volume snapshot and cloning | Able to take consistent snapshots across all available replicas of a volume and restore to a given snapshot | Replicated PV Mayastor | v4.1 (Q3 2024) | Completed |
+| Local PV CI | CI hardening and enhancements, helm chart support and more tests | Local PV (LVM, ZFS, Hostpath) | v4.2 (Q4 2024) | In progress |
+| Local PV E2E | E2E hardening, umbrella chart testing, conversion of Ansible to Ginkgo-based BDDs | Local PV (LVM, ZFS, Hostpath) | v4.2 (Q4 2024) | In progress |
+| Replica topology | Replica distribution based on pool and node topologies | Replicated PV Mayastor | v4.2 (Q4 2024) | In progress |
+| NVMe-oF over RDMA support | Support for NVMe-oF over RDMA as transport | Replicated PV Mayastor | v4.2 (Q4 2024) | In progress |
+| One kubectl plug-in | Unified kubectl plug-in to manage all OpenEBS components | All | v4.2 (Q4 2024) | In progress |
```
Comment (blaisedias) [inline ROADMAP.md:29]: nit: One kubectl plug-in -> Single kubectl plugin

**Review by blaisedias — APPROVED** [review]

File: ROADMAP.md (hunk 3 above)
Comment (tiagolobocastro) [reply ROADMAP.md:29]: unified?

File: ROADMAP.md
[the remark this answers is not in the document]
Comment (avishnu) [reply ROADMAP.md:1]: Can you point me to this? I saw one instance which I modified in the PR.

**Review by Abhinandan-Purkait — APPROVED** [review]

File: ROADMAP.md
[the remark this answers is not in the document]
Comment (tiagolobocastro) [reply ROADMAP.md:1]: Line 53:
[Joining our Community meetings](https://github.com/openebs/openebs/tree/master/community)

**Review by tiagolobocastro — APPROVED** [review]

File: ROADMAP.md (hunk 1 above)
Comment (Abhinandan-Purkait) [reply ROADMAP.md:20]: How about using the term Unified for Installer, documentation and the plugin.

**Review by niladrih — APPROVED** [review]

File: ROADMAP.md
[the remark this answers is not in the document]
Comment (avishnu) [reply ROADMAP.md:1]: Done

File: ROADMAP.md (hunk 1 above)
Comment (avishnu) [reply ROADMAP.md]: Done

File: ROADMAP.md (hunk 3 above)
Comment (avishnu) [reply ROADMAP.md]: Done

File: ROADMAP.md (hunk 2 above)
Comment (avishnu) [reply ROADMAP.md]: Done

File: ROADMAP.md [hunk 4]
```diff
@@ -6,96 +6,49 @@ Community and contributor involvement is vital for successfully implementing all
...
+| Volume resize | Able to increase volume size and overlaying filesystem size with I/O continuity | Replicated PV Mayastor | v4.0 (Q1 2024) | Completed |
+| Multi-replica volume snapshot and cloning | Able to take consistent snapshots across all available replicas of a volume and restore to a given snapshot | Replicated PV Mayastor | v4.1 (Q3 2024) | Completed |
+| Local PV CI | CI hardening and enhancements, helm chart support and more tests | Local PV (LVM, ZFS, Hostpath) | v4.2 (Q4 2024) | In progress |
+| Local PV E2E | E2E hardening, umbrella chart testing, conversion of Ansible to Ginkgo-based BDDs | Local PV (LVM, ZFS, Hostpath) | v4.2 (Q4 2024) | In progress |
+| Replica topology | Replica distribution based on pool and node topologies | Replicated PV Mayastor | v4.2 (Q4 2024) | In progress |
+| NVMe-oF over RDMA support | Support for NVMe-oF over RDMA as transport | Replicated PV Mayastor | v4.2 (Q4 2024) | In progress |
+| Unified kubectl plugin | Unified kubectl plugin to manage all OpenEBS components | All | v4.2 (Q4 2024) | In progress |
+| Local PV LVM cloning | Able to do K8s restore of Local PV LVM snapshot | Local PV LVM | v4.3 (Q1 2025) | In progress |
```
Comment (dsharma-dc) [inline ROADMAP.md:30]: Some upcoming items are marked `In-progress` and some have status blank. `In-progress` doesn't seem right status for these.

Comment (avishnu) [reply ROADMAP.md:30]: The upcoming items marked as in-progress already have ongoing efforts. The blank ones haven't been started. Hope that explains.

Comment (avishnu) [reply ROADMAP.md]: Fixed the status on Local PV LVM cloning feature.

**Review by orville-wright — APPROVED** [review]
Looks good. - Approved

**Review by rohan2794 — APPROVED** [review]

**Review by dsharma-dc — APPROVED** [review]

**Pull request merged**